h2o-automl/src/test/java/ai/h2o/automl/EventLogTest.java

```diff
@@ -0,0 +1,58 @@
+package ai.h2o.automl;
+
+import ai.h2o.automl.EventLogEntry.Level;
+import ai.h2o.automl.EventLogEntry.Stage;
+import org.apache.commons.lang.time.DateUtils;
+import org.joda.time.DateTime;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import water.Key;
+import water.util.TwoDimTable;
+
+import java.util.Calendar;
+import java.util.Date;
+
+public class EventLogTest extends water.TestUtil {
+
+    @BeforeClass
+    public static void setup() {
+        stall_till_cloudsize(1);
+    }
+
+    private static Key<AutoML> dummy = Key.make();
+
+    @Test public void test_events_are_recorded_as_expected() throws Exception {
+        EventLog eventLog = new EventLog(dummy);
+        try {
+            eventLog.debug(Stage.Workflow, "debug workflow");
+            eventLog.info(Stage.Workflow, "info workflow");
+            eventLog.warn(Stage.ModelTraining, "warn training");
+            eventLog.info(Stage.ModelTraining, "this means what it is").setNamedValue("foo", "bar");
+            eventLog.info(Stage.ModelTraining, "this doesn't mean what it looks").setNamedValue("bar", 777);
+
+            TwoDimTable events = eventLog.toTwoDimTable("Test header");
+            Assert.assertEquals(6, events.getColDim());
```

Comment:
@seb-h2o maybe it is unnecessary to check number of columns as you are checking it with headers afterwards.

Reply:
true, but I started with dims indeed :) can't hurt